modules/freewheelSSPBidAdapter.js

```diff
@@ -0,0 +1,314 @@
+    if (config.hasOwnProperty(key) && key !== 'format' && key !== 'zone' && key !== 'zoneId' && key !== 'vastUrlParams') {
+      script += ',' + key + ':"' + config[key] + '"';
+    }
+  }
+  script += '};' +
+
+  'window.com.stickyadstv.' + getAPIName(bid.params.format) + '.start(config);';
+
+  return script;
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: ['video'],
+  aliases: ['stickyadstv'], //  former name for freewheel-ssp
+  /**
+  * Determines whether or not the given bid request is valid.
+  *
+  * @param {object} bid The bid to validate.
+  * @return boolean True if this is a valid bid, and false otherwise.
+  */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.zoneId);
+  },
+
+  /**
+  * Make a server request from the list of BidRequests.
+  *
+  * @param {BidRequest[]} bidRequests A non-empty list of bid requests which should be sent to the Server.
+  * @return ServerRequest Info describing the request to the server.
+  */
+  buildRequests: function(bidRequests) {
+    // var currency = config.getConfig(currency);
+
+    var currentBidRequest = bidRequests[0];
+    if (bidRequests.length > 1) {
+      utils.logMessage('Prebid.JS - freewheel bid adapter: only one ad unit is required.');
+    }
+
+    var requestParams = {
+      reqType: 'AdsSetup',
+      protocolVersion: '2.0',
+      zoneId: currentBidRequest.params.zoneId,
+      componentId: getComponentId(currentBidRequest.params.format)
+    };
+
+    var location = utils.getTopWindowUrl();
+    if (isValidUrl(location)) {
+      requestParams.loc = location;
+    }
+
+    var playerSize = getBiggerSize(currentBidRequest.sizes);
+    if (playerSize[0] > 0 || playerSize[1] > 0) {
+      requestParams.playerSize = playerSize[0] + 'x' + playerSize[1];
+    }
+
+    return {
+      method: 'GET',
+      url: FREEWHEEL_ADSSETUP,
+      data: requestParams,
+      bidRequest: currentBidRequest
+    };
+  },
+
+  /**
+  * Unpack the response from the server into a list of bids.
+  *
+  * @param {*} serverResponse A successful response from the server.
+  * @param {object} request: the built request object containing the initial bidRequest.
+  * @return {Bid[]} An array of bids which were nested inside the server.
+  */
+  interpretResponse: function(serverResponse, request) {
+    var bidrequest = request.bidRequest;
+    var playerSize = getBiggerSize(bidrequest.sizes);
+
+    if (typeof serverResponse == 'object' && typeof serverResponse.body == 'string') {
+      serverResponse = serverResponse.body;
+    }
+
+    var xmlDoc;
+    try {
+      var parser = new DOMParser();
+      xmlDoc = parser.parseFromString(serverResponse, 'application/xml');
+    } catch (err) {
+      utils.logWarn('Prebid.js - ' + BIDDER_CODE + ' : ' + err);
+      return;
+    }
+
+    const princingData = getPricing(xmlDoc);
+    const creativeId = getCreativeId(xmlDoc);
+
+    const topWin = getTopMostWindow();
+    if (!topWin.freeheelssp_cache) {
```

Comment:
Is it just me or is `freeheel` a consistent typo?

Reply:
Indeed!   As you imagine it should be "freewheelssp".
Thx, I'll fix it.

Reply:
done.